aea/protocols/oef/message.py

```diff
@@ -76,67 +76,104 @@ def __init__(self, oef_type: Optional[Type] = None,
-                assert self.is_set("service_id")
-                service_description = self.get("service_description")
-                service_id = self.get("service_id")
-                assert isinstance(service_description, Description)
-                assert isinstance(service_id, str)
-            elif oef_type == OEFMessage.Type.REGISTER_AGENT:
-                assert self.is_set("id")
-                assert self.is_set("agent_description")
-                assert self.is_set("agent_id")
-                agent_description = self.get("agent_description")
-                agent_id = self.get("agent_id")
-                assert isinstance(agent_description, Description)
-                assert isinstance(agent_id, str)
-            elif oef_type == OEFMessage.Type.UNREGISTER_SERVICE:
-                assert self.is_set("id")
-                assert self.is_set("service_description")
-                assert self.is_set("service_id")
-                service_description = self.get("service_description")
-                service_id = self.get("service_id")
-                assert isinstance(service_description, Description)
-                assert isinstance(service_id, str)
-            elif oef_type == OEFMessage.Type.UNREGISTER_AGENT:
-                assert self.is_set("id")
-                assert self.is_set("agent_description")
-                assert self.is_set("agent_id")
-                agent_description = self.get("agent_description")
-                agent_id = self.get("agent_id")
-                assert isinstance(agent_description, Description)
-                assert isinstance(agent_id, str)
-            elif oef_type == OEFMessage.Type.SEARCH_SERVICES:
-                assert self.is_set("id")
-                assert self.is_set("query")
-                query = self.get("query")
-                assert isinstance(query, Query)
-            elif oef_type == OEFMessage.Type.SEARCH_AGENTS:
-                assert self.is_set("id")
-                assert self.is_set("query")
-                query = self.get("query")
-                assert isinstance(query, Query)
-            elif oef_type == OEFMessage.Type.SEARCH_RESULT:
-                assert self.is_set("id")
-                assert self.is_set("agents")
-                agents = cast(List[str], self.get("agents"))
-                assert type(agents) == list and all(type(a) == str for a in agents)
-            elif oef_type == OEFMessage.Type.OEF_ERROR:
-                assert self.is_set("id")
-                assert self.is_set("operation")
-                operation = self.get("operation")
-                assert operation in set(OEFMessage.OEFErrorOperation)
-            elif oef_type == OEFMessage.Type.DIALOGUE_ERROR:
-                assert self.is_set("id")
-                assert self.is_set("dialogue_id")
-                assert self.is_set("origin")
+            assert self.type is not None, "type must not be None."
```

Comment:
same as above, I don't think we can use properties in asserts here as this defeats the point of the checks